src/map/mob.cpp

```diff
@@ -316,31 +317,41 @@ std::shared_ptr<s_mob_db> mobdb_search_aegisname( const char* str ){
 }
 
 /*==========================================
- * Searches up to N matches. Returns number of matches [Skotlex]
+ * Searches up to N matches. Prioritizing full matches first. Returns number of matches [Skotlex]
  *------------------------------------------*/
-uint16 mobdb_searchname_array_(const char *str, uint16 * out, uint16 size, bool full_cmp)
+uint16 mobdb_searchname_array(const char *str, uint16 * out, uint16 size)
 {
 	uint16 count = 0;
 	const auto &mob_list = mob_db.getCache();
 
+	// Full compare first
 	for( const auto &mob : mob_list ) {
 		if (mob == nullptr)
 			continue;
-		if( mobdb_searchname_sub(mob->id, str, full_cmp) ) {
+		if( mobdb_searchname_sub(mob->id, str, true) ) {
 			if( count < size )
 				out[count] = mob->id;
```

Comment:
```suggestion
			out[count] = mob->id;
```
Since you cancel the loop now.